plan/logical_plans.go

```diff
@@ -146,6 +146,18 @@ type Selection struct {
 
 	// onTable means if this selection's child is a table scan or index scan.
 	onTable bool
+
+	// If ScanController is true, then the child of this selection is a scan,
+	// which use pk or index. we will record the accessConditions, idxConditions,
+	// and tblConditions to control the below plan.
+	ScanController bool
+
+	// Since one selection may call convert2PhysicalScan many times. We extract the PkName and indices
+	// used for scanController only once and store them to judge whether this selection can convert to
+	// scanController mode.
+	usefulPkName         model.CIStr
+	usefulIndices        []*model.IndexInfo
+	extractedUsefulThing bool
```

Comment:
set a flag that we set controller when the flag is true. The flag is true only the selection has correlated column and its child is DataSource. You can check this during decorrelation.